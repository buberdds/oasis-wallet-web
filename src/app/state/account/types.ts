import { Transaction } from 'app/state/transaction/types'
import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export interface BalanceDetails {
  available: StringifiedBigInt | null
  debonding: StringifiedBigInt | null
  delegations: StringifiedBigInt | null
  total: StringifiedBigInt | null
}

export interface Allowance {
  address: string
  amount: StringifiedBigInt
}

export interface Account extends BalanceDetails {
  address: string
  allowances?: Allowance[]
}

/* --- STATE --- */
export interface AccountState extends Account {
  loading: boolean
  accountError?: ErrorPayload
  transactions: Transaction[]
  transactionsError?: ErrorPayload
}
