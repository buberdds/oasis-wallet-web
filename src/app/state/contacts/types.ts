export interface Contact {
  address: string
  name: string
}

/* --- STATE --- */
export interface ContactsState {
  [key: string]: Contact
}
