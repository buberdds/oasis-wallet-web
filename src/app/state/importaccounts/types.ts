import { ErrorPayload } from 'types/errors'
import { WalletBalance, WalletType } from '../wallet/types'

/* --- STATE --- */
export interface ImportAccountsListAccount {
  address: string
  balance: WalletBalance
  path: number[]
  privateKey?: string
  publicKey: string
  selected: boolean
  type: WalletType
}

export enum ImportAccountsStep {
  OpeningUSB = 'opening_usb',
  LoadingAccounts = 'loading_accounts',
  LoadingBalances = 'loading_balances',
  Done = 'done',
}

export interface ImportAccountsState {
  accounts: ImportAccountsListAccount[]
  deviceRequested: boolean
  error?: ErrorPayload
  showAccountsSelectionModal: boolean
  step?: ImportAccountsStep
}
