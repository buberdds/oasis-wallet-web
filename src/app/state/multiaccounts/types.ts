import { ErrorPayload } from 'types/errors'
import { WalletBalance } from '../wallet/types'

/* --- STATE --- */
export interface MultiAccountsListAccount {
  publicKey: string
  address: string
  path: number[]
  balance: WalletBalance
  selected: boolean
}

export enum MultiAccountsStep {
  OpeningUSB = 'opening_usb',
  LoadingAccounts = 'loading_accounts',
  LoadingBalances = 'loading_balances',
  Done = 'done',
}

export interface MultiAccountsState {
  accounts: MultiAccountsListAccount[]
  error?: ErrorPayload
  mnemonic?: string
  step?: MultiAccountsStep
}
