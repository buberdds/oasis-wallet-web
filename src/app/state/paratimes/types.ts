import { StringifiedBigInt } from 'types/StringifiedBigInt'
import { ErrorPayload } from 'types/errors'
import { ParaTime } from '../../../config'

export enum TransactionFormSteps {
  TransferType = 'transferType',
  ParaTimeSelection = 'paraTimeSelection',
  TransactionRecipient = 'transactionRecipient',
  TransactionAmount = 'transactionAmount',
  TransactionConfirmation = 'transactionConfirmation',
  TransactionSummary = 'transactionSummary',
  TransactionError = 'transactionError',
}

export enum TransactionTypes {
  Deposit = 'deposit',
  Withdraw = 'withdraw',
}

export interface TransactionForm {
  amount: string
  confirmation: boolean
  feeAmount: string
  feeGas: string
  paraTime?: ParaTime
  privateKey: string
  recipient: string
  type: TransactionTypes | undefined
}

export interface ParaTimesState {
  balance: StringifiedBigInt
  isLoading: boolean
  transactionError?: ErrorPayload
  transactionForm: TransactionForm
  transactionFormStep: TransactionFormSteps
}

export type Runtime = {
  address: string
  id: string
  decimals: number
}

export type ParaTimeTransaction = Pick<
  TransactionForm,
  'amount' | 'privateKey' | 'feeAmount' | 'feeGas' | 'recipient' | 'type'
>
