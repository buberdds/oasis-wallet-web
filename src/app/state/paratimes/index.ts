import { PayloadAction } from '@reduxjs/toolkit'
import { StringifiedBigInt } from 'types/StringifiedBigInt'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { ParaTimesState, TransactionForm, TransactionFormSteps, TransactionTypes } from './types'

export const initialState: ParaTimesState = {
  balance: '',
  isLoading: false,
  transactionForm: {
    amount: '',
    confirmation: false,
    paraTime: undefined,
    privateKey: '',
    recipient: '',
    type: undefined,
  },
  transactionFormStep: TransactionFormSteps.TransferType,
}

const slice = createSlice({
  name: 'paraTimes',
  initialState,
  reducers: {
    balanceLoaded(state, action: PayloadAction<StringifiedBigInt>) {
      state.balance = action.payload
      state.isLoading = false
    },
    fetchBalanceUsingEthPrivateKey(state, action: PayloadAction<void>) {
      state.isLoading = true
    },
    fetchBalanceUsingOasisAddress(state, action: PayloadAction<void>) {
      state.isLoading = true
    },
    resetTransactionForm(state, action: PayloadAction<void>) {
      state.transactionForm = initialState.transactionForm
      state.transactionFormStep = TransactionFormSteps.TransferType
    },
    navigateToDeposit(state, action: PayloadAction<void>) {
      state.transactionForm.type = TransactionTypes.Deposit
      state.transactionFormStep = TransactionFormSteps.ParaTimeSelection
    },
    navigateToWithdraw(state, action: PayloadAction<void>) {
      state.transactionForm.type = TransactionTypes.Withdraw
      state.transactionFormStep = TransactionFormSteps.ParaTimeSelection
    },
    navigateToRecipient(state, action: PayloadAction<void>) {
      state.transactionFormStep = TransactionFormSteps.TransactionRecipient
    },
    navigateToAmount(state, action: PayloadAction<void>) {
      state.transactionFormStep = TransactionFormSteps.TransactionAmount
    },
    navigateToParaTimes(state, action: PayloadAction<void>) {
      state.transactionFormStep = TransactionFormSteps.TransferType
    },
    navigateToConfirmation(state, action: PayloadAction<void>) {
      state.transactionFormStep = TransactionFormSteps.TransactionConfirmation
    },
    navigateToSummary(state, action: PayloadAction<void>) {
      state.transactionFormStep = TransactionFormSteps.TransactionSummary
    },
    setTransactionForm(state, action: PayloadAction<TransactionForm>) {
      state.transactionForm = action.payload
    },
  },
})

export const { actions: paraTimesActions } = slice

export default slice.reducer
