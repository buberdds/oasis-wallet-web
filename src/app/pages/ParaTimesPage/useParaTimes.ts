import { useCallback } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useTranslation } from 'react-i18next'
import { paraTimesActions } from 'app/state/paratimes'
import { TransactionForm, TransactionTypes } from 'app/state/paratimes/types'
import { selectSelectedNetwork, selectTicker } from 'app/state/network/selectors'
import { selectAccountAvailableBalance, selectAccountIsLoading } from 'app/state/account/selectors'
import { selectAddress } from 'app/state/wallet/selectors'
import { selectParaTimes } from 'app/state/paratimes/selectors'
import { paraTimesConfig, RuntimeTypes, ParaTime } from '../../../config'

type AvailableParaTimesForNetwork = {
  isEvm: boolean
  value: ParaTime
}

const evmcParaTimes = Object.keys(paraTimesConfig).filter(
  key => paraTimesConfig[key as ParaTime].type === RuntimeTypes.Evm,
)

export const useParaTimes = () => {
  const { t } = useTranslation()
  const dispatch = useDispatch()
  const resetTransactionForm = useCallback(() => {
    dispatch(paraTimesActions.resetTransactionForm())
  }, [dispatch])
  const setTransactionForm = (formValues: TransactionForm) =>
    dispatch(paraTimesActions.setTransactionForm(formValues))
  const { balance, isLoading, transactionForm } = useSelector(selectParaTimes)
  const accountBalance = useSelector(selectAccountAvailableBalance)
  const accountIsLoading = useSelector(selectAccountIsLoading)
  const accountAddress = useSelector(selectAddress)
  const selectedNetwork = useSelector(selectSelectedNetwork)
  const ticker = useSelector(selectTicker)
  const isDepositing = transactionForm.type !== TransactionTypes.Withdraw
  const isEvmcParaTime = evmcParaTimes.includes(transactionForm.paraTime!)
  const needsEthAddress = isDepositing && isEvmcParaTime
  const balanceInBaseUnit = isDepositing || (!isDepositing && !isEvmcParaTime)
  const paraTimeName = transactionForm.paraTime ? t(`paraTimes.common.${transactionForm.paraTime}`) : ''
  const availableParaTimesForSelectedNetwork: AvailableParaTimesForNetwork[] = (
    Object.keys(paraTimesConfig) as ParaTime[]
  )
    .filter(paraTimeKey => paraTimesConfig[paraTimeKey][selectedNetwork].runtimeId)
    .map(paraTimeKey => ({
      isEvm: paraTimesConfig[paraTimeKey].type === RuntimeTypes.Evm,
      value: paraTimeKey,
    }))
  const walletBalance = !isDepositing ? balance : accountBalance

  return {
    accountAddress,
    accountIsLoading,
    availableParaTimesForSelectedNetwork,
    balance: walletBalance,
    balanceInBaseUnit,
    isDepositing,
    isWalletEmpty: walletBalance === '0',
    isEvmcParaTime,
    isLoading,
    paraTimeName,
    resetTransactionForm,
    ticker,
    transactionForm,
    setTransactionForm,
    usesOasisAddress: !needsEthAddress,
  }
}
