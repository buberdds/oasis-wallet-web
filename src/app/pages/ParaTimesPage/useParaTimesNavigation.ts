import { useCallback } from 'react'
import { useDispatch } from 'react-redux'
import { paraTimesActions } from 'app/state/paratimes'

export const useParaTimesNavigation = () => {
  const dispatch = useDispatch()

  const navigateToDeposit = useCallback(() => dispatch(paraTimesActions.navigateToDeposit()), [dispatch])
  const navigateToWithdraw = useCallback(() => dispatch(paraTimesActions.navigateToWithdraw()), [dispatch])
  const navigateToParaTimes = useCallback(() => dispatch(paraTimesActions.navigateToParaTimes()), [dispatch])
  const navigateToRecipient = useCallback(() => dispatch(paraTimesActions.navigateToRecipient()), [dispatch])
  const navigateToAmount = useCallback(() => dispatch(paraTimesActions.navigateToAmount()), [dispatch])
  const navigateToConfirmation = useCallback(
    () => dispatch(paraTimesActions.navigateToConfirmation()),
    [dispatch],
  )
  const navigateToSummary = useCallback(() => dispatch(paraTimesActions.navigateToSummary()), [dispatch])

  return {
    navigateToAmount,
    navigateToDeposit,
    navigateToParaTimes,
    navigateToRecipient,
    navigateToConfirmation,
    navigateToSummary,
    navigateToWithdraw,
  }
}
