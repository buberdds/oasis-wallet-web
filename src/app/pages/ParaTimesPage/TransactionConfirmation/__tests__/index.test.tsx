import * as React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { TransactionForm, TransactionTypes } from 'app/state/paratimes/types'
import { useParaTimes, ParaTimesHook } from '../../useParaTimes'
import { useParaTimesNavigation, ParaTimesNavigationHook } from '../../useParaTimesNavigation'
import { TransactionConfirmation } from '..'

jest.unmock('react-i18next')
jest.mock('../../useParaTimes')
jest.mock('../../useParaTimesNavigation')

describe('<TransactionConfirmation />', () => {
  const mockUseParaTimesResult = {
    isDepositing: true,
    isEvmcParaTime: false,
    paraTimeName: 'Cipher',
    ticker: 'ROSE',
    transactionForm: {
      amount: '10',
      recipient: 'dummyAddress',
    },
  } as ParaTimesHook
  const mockUseParaTimesNavigationResult = {} as ParaTimesNavigationHook

  beforeEach(() => {
    jest.mocked(useParaTimes).mockReturnValue(mockUseParaTimesResult)
    jest.mocked(useParaTimesNavigation).mockReturnValue(mockUseParaTimesNavigationResult)
  })

  it('should render component', () => {
    const { container } = render(<TransactionConfirmation />)

    expect(container).toMatchSnapshot()
  })

  it('should render EVMc withdraw variant component', () => {
    jest.mocked(useParaTimes).mockReturnValue({
      isDepositing: false,
      isEvmcParaTime: true,
      paraTimeName: 'Emerald',
      ticker: 'ROSE',
      transactionForm: {
        amount: '10',
        recipient: 'dummyAddress',
        type: TransactionTypes.Withdraw,
      },
    } as ParaTimesHook)
    render(<TransactionConfirmation />)

    expect(screen.getByTestId('paraTime-content-description')).toMatchSnapshot()
  })

  it('should require a confirmation on form submit', () => {
    const navigateToSummary = jest.fn()
    jest
      .mocked(useParaTimesNavigation)
      .mockReturnValue({ ...mockUseParaTimesNavigationResult, navigateToSummary })
    render(<TransactionConfirmation />)

    userEvent.click(screen.getByRole('button', { name: 'Deposit' }))

    expect(screen.getByText('Field is required')).toBeInTheDocument()
    expect(navigateToSummary).not.toHaveBeenCalled()
  })

  it('should allow to navigate to summary section', () => {
    const navigateToSummary = jest.fn()
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      transactionForm: {
        amount: '10',
        confirmation: true,
        recipient: 'dummyAddress',
        type: TransactionTypes.Withdraw,
      } as TransactionForm,
    })
    jest.mocked(useParaTimesNavigation).mockReturnValue({
      ...mockUseParaTimesNavigationResult,
      navigateToSummary,
    })
    render(<TransactionConfirmation />)

    userEvent.click(screen.getByRole('button', { name: 'Deposit' }))

    expect(navigateToSummary).toHaveBeenCalled()
  })

  it('should navigate back to amount selection step', () => {
    const navigateToAmount = jest.fn()
    jest.mocked(useParaTimesNavigation).mockReturnValue({
      ...mockUseParaTimesNavigationResult,
      navigateToAmount,
    })
    render(<TransactionConfirmation />)

    userEvent.click(screen.getByRole('button', { name: 'Previous' }))

    expect(navigateToAmount).toHaveBeenCalled()
  })
})
