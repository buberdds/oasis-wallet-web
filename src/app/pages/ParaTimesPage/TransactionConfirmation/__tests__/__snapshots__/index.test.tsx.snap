// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionConfirmation /> should render EVMc withdraw variant component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 48px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
  text-align: center;
}

@media only screen and (max-width:768px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

<div
  class="c0"
  data-testid="paraTime-content-description"
  style="max-width: 550px;"
>
  <span
    class="c1"
  >
    You are about to transfer 
    <strong>
      10 ROSE
    </strong>
     tokens from the withdrawing wallet on the 
    <strong>
      Emerald
    </strong>
     (EVMc) ParaTime: 
    <strong>
      dummyAddress
    </strong>
  </span>
</div>
`;

exports[`<TransactionConfirmation /> should render component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 48px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 96px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 24px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 2px rgba(0,0,0,0.15);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 24px;
  width: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  height: 24px;
}

.c3 {
  font-size: 18px;
  line-height: 24px;
  text-align: center;
}

.c17 {
  font-size: 14px;
  line-height: 20px;
}

.c16 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  width: 100%;
}

.c16:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c16:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c16:focus > circle,
.c16:focus > ellipse,
.c16:focus > line,
.c16:focus > path,
.c16:focus > polygon,
.c16:focus > polyline,
.c16:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c16:focus::-moz-focus-inner {
  border: 0;
}

.c16:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c16:focus:not(:focus-visible) > circle,
.c16:focus:not(:focus-visible) > ellipse,
.c16:focus:not(:focus-visible) > line,
.c16:focus:not(:focus-visible) > path,
.c16:focus:not(:focus-visible) > polygon,
.c16:focus:not(:focus-visible) > polyline,
.c16:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c16:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c1 {
  margin-bottom: 48px;
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

.c10:hover input:not([disabled]) + div,
.c10:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c13 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c13:checked + span > span {
  left: calc( 48px - 24px );
  background: #7D4CDB;
}

.c12 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    margin-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c14 {
    border: solid 2px rgba(0,0,0,0.15);
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

<div>
  <div
    class="c0"
  >
    <h3
      class="c1"
    >
      Deposit to ParaTime
    </h3>
    <div
      class="c2"
      data-testid="paraTime-content-description"
      style="max-width: 550px;"
    >
      <span
        class="c3"
      >
        You are about to transfer 
        <strong>
          10 ROSE
        </strong>
         tokens into the receiving wallet on the 
        <strong>
          Cipher
        </strong>
          ParaTime: 
        <strong>
          dummyAddress
        </strong>
      </span>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
        style="max-width: 450px;"
      >
        <span
          class="c3"
        >
          Please confirm the transferring amount and the receiving wallet's address are correct and then click "Deposit" to make the transfer.
        </span>
      </div>
      <div
        class="c6"
      />
      <form>
        <div
          class="c7"
        >
          <div
            class="c8 "
          >
            <div
              class="c9 FormField__FormFieldContentBox-sc-m9hood-1"
            >
              <label
                class="c10"
                label="I confirm the amount and the address are correct"
              >
                <div
                  class="c11 c12"
                >
                  <input
                    class="c13"
                    name="confirmation"
                    type="checkbox"
                  />
                  <div
                    class="c14 "
                  />
                </div>
                <span>
                  I confirm the amount and the address are correct
                </span>
              </label>
            </div>
          </div>
        </div>
        <div
          class="c15"
        >
          <div
            class="c5"
          >
            <button
              class="c16"
              type="submit"
            >
              Deposit
            </button>
          </div>
        </div>
        <span
          class="c17"
        >
          * EVMc - Ethereum Virtual Machine compatible
        </span>
      </form>
    </div>
  </div>
</div>
`;
